extension/src/main/java/com/cliqz/extension/CliqzExtensionFeature.kt

```diff
@@ -0,0 +1,142 @@
+package com.cliqz.extension
+
+import kotlinx.coroutines.*
+import mozilla.components.browser.engine.gecko.webextension.GeckoWebExtension
+import mozilla.components.concept.engine.EngineSession
+import mozilla.components.concept.engine.webextension.MessageHandler
+import mozilla.components.concept.engine.webextension.Port
+import mozilla.components.support.base.log.logger.Logger
+import org.json.JSONArray
+import org.json.JSONObject
+import org.mozilla.geckoview.GeckoRuntime
+import java.util.concurrent.ConcurrentHashMap
+import java.util.concurrent.CountDownLatch
+
+val DEMOGRAPHICS = mapOf(
+        "brand" to "cliqz",
+        "name" to "browser",
+        "platform" to "android"
+)
+val TELEMETRY_CHANNEL = if (BuildConfig.DEBUG) "MA99" else "MA60"
+
+/**
+ * @author Sam Macbeth
+ */
+class CliqzExtensionFeature(val runtime: GeckoRuntime) {
+    private val logger = Logger("cliqz-extension")
+
+    private val appName = "cliqz"
+    private val privacyExtensionID = "cliqz@cliqz.com"
+    private val privacyExtensionUrl = "resource://android/assets/extensions/cliqz/"
+
+    private var extension = GeckoWebExtension(privacyExtensionID, privacyExtensionUrl, runtime.webExtensionController, allowContentMessaging = true)
+    private val messageHandler = CliqzBackgroundMessageHandler(this)
+
+    val extensionConfig: JSONObject by lazy {
+        val settings = JSONObject(mapOf(
+                "channel" to TELEMETRY_CHANNEL,
+                "telemetry" to mapOf(
+                        "demographics" to DEMOGRAPHICS
+                ),
+                "ADBLOCKER_PLATFORM" to "desktop"
```

Comment:
is this correct?

Reply:
See https://github.com/cliqz/navigation-extension/pull/8798#discussion_r393490370